contracts/DelegateContribute.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+interface IWVEOLAS {
+    /// @dev Gets the account balance in native token.
+    /// @param account Account address.
+    /// @return balance Account balance.
+    function balanceOf(address account) external view returns (uint256 balance);
+
+    /// @dev Gets the voting power.
+    /// @param account Account address.
+    /// @return balance Account balance.
+    function getVotes(address account) external view returns (uint256 balance);
+}
+
+/// @dev No balance available to delegate.
+/// @param delegator provided delegator address.
+error NoBalance(address delegator);
+
+/// @title DynamicContribution - Ownable smart contract for minting ERC721 tokens
+contract DelegateContribute {
+
+    /// @dev Delegation.
+    /// @param delegator provided delegator.
+    /// @param delegatee provided delegatee.
+    event Delegation(address indexed delegator, address indexed delegatee);
+
+    // wveOLAS address
+    address immutable wveOLAS;
+
+    // Mapping of account address => address
+    mapping(address => address) public mapDelegation;
+
+    // Maps a delegatee to their list of delegators
+    mapping(address => address[]) private delegatorLists;
+
+    // Maps a delegator to their index in the delegatee's list + 1 (since default value is 0)
+    mapping(address => mapping(address => uint256)) private delegatorIndex;
+
+    /// @dev DynamicContribution constructor.
+    /// @param _wveOLAS wveOLAS contract address.
+    constructor(address _wveOLAS)
+    {
+        wveOLAS = _wveOLAS;
+    }
+
+    /// @dev Delegates Contribute voting power to an address.
+    /// @param delegator Delegator address.
+    /// @param delegatee Delegatee address.
+    function delegate(address delegator, address delegatee) external {
```

Comment:
This delegation method handles redelegation and undelegation

Reply:
It does not check any ownership or access, basically anybody can delegate for a specific delegator. Is this intentional?

Reply:
lol - my bad. I did this late in the evening.